### PR TITLE
editor: fix unstranslated label for hidden fields

### PR DESCRIPTION
**Pull request by jma**
* Uses ngx formly 5.9.
* Makes the autocomplete component asynchronous to allows hot language
switch.
* Tuncates long labels in the add field components.
* Fixes autocomplete translations.
* Fixes field dropdown menu translations.
* Removes useless function and code.
* Fixes cssClass for multischema object.
* Moves the indentation value into the css instead of using bootstrap
classes.
* Fixes scroll and focus action when a new field is added or when a link
in the navigation is clicked.
* Stores the untranslated label in the formly field.
* Removes extra border for multischema element.
    
Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- https://tree.taiga.io/project/rero21-reroils/us/1542?milestone=269576
- Fixes untranslated labels in the navigations and dropdown elements.

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/rero/ng-core/src/lib/record/editor/add-field-editor/add-field-editor.component.ts [hunk 1]
```diff
@@ -17,17 +17,10 @@
...
 import { EditorService } from '../editor.service';
-
-/**
- * Type a head selection type
- */
-interface FormlyFieldConfigSelection {
-  name: any;
-  field: FormlyFieldConfig;
-}
+import { TranslateService } from '@ngx-translate/core';
```
Comment (AoNoOokami) [inline projects/rero/ng-core/src/lib/record/editor/add-field-editor/add-field-editor.component.ts:23]: Did you organize imports?

File: projects/rero/ng-core/src/lib/record/editor/add-field-editor/add-field-editor.component.ts [hunk 2]
```diff
@@ -37,50 +30,73 @@ interface FormlyFieldConfigSelection {
...
-
-  // list of fields to populate autocomplete
-  @Input()
-  fields: FormlyFieldConfig[];
+  typeaheadFields$: Observable<Array<FormlyFieldConfig>>;
 
   /***
    * Constructor
    * @param _editorService - EditorService, that keep the list of hidden fields
    */
```
Comment (AoNoOokami) [inline projects/rero/ng-core/src/lib/record/editor/add-field-editor/add-field-editor.component.ts:43]: Missing docstring for translateService

File: projects/rero/ng-core/src/lib/record/editor/add-field-editor/add-field-editor.component.ts [hunk 3]
```diff
@@ -37,50 +30,73 @@ interface FormlyFieldConfigSelection {
...
+      .pipe(
+        mergeMap((token: string) => this.getSuggestionsList(token))
+      );
+  }
+
+  /**
+   * Generate the suggestion list for the autocomplete component.
+   *
+   * @param token string to filter the autocomplete list
+   */
```
Comment (AoNoOokami) [inline projects/rero/ng-core/src/lib/record/editor/add-field-editor/add-field-editor.component.ts:61]: I think you should precise the @return in the docstring

File: projects/rero/ng-core/src/lib/record/editor/add-field-editor/add-field-editor.component.ts [hunk 4]
```diff
@@ -37,50 +30,73 @@ interface FormlyFieldConfigSelection {
...
-        return value;
-      })
-    );
   }
 
+  /**
+   * Translate the label of a given formly field.
+   *
+   * @param field ngx-formly field
+   */
```
Comment (AoNoOokami) [inline projects/rero/ng-core/src/lib/record/editor/add-field-editor/add-field-editor.component.ts:89]: @return too

File: projects/rero/ng-core/src/lib/record/editor/horizontal-wrapper/horizontal-wrapper.component.ts [hunk 5]
```diff
@@ -22,9 +22,8 @@ import { FieldWrapper } from '@ngx-formly/core';
   template: `
     <div class="form-group m-0">
       <div class="d-flex">
-        <label [attr.for]="id" class="text-nowrap mr-2 col-form-label-1" *ngIf="to.label" [tooltip]="to.description">
-          {{ to.label }}
-          <ng-container *ngIf="to.required && to.hideRequiredMarker !== true">*</ng-container>
+        <label [attr.for]="id" class="text-nowrap1 mr-2 col-form-label-1" *ngIf="to.label" [tooltip]="to.description">
+          {{ to.label }}<ng-container *ngIf="to.required && to.hideRequiredMarker !== true">&nbsp;*</ng-container>
```
Comment (zannkukai) [inline projects/rero/ng-core/src/lib/record/editor/horizontal-wrapper/horizontal-wrapper.component.ts:26]: Cosmeic/For a next PR : I think the "required star" should be highlighted with red and/or bold font.

Comment (jma) [reply projects/rero/ng-core/src/lib/record/editor/horizontal-wrapper/horizontal-wrapper.component.ts:26]: > Cosmetic/For a next PR : I think the "required star" should be highlighted with red and/or bold font.

This should need to be discussed with the PO. Thanks for your fast review.

**Pull request merged**